ngrinder-controller/src/main/java/org/ngrinder/infra/config/WebMvcConfig.java

```diff
@@ -0,0 +1,154 @@
+import org.springframework.web.servlet.view.freemarker.FreeMarkerViewResolver;
+
+import javax.servlet.ServletContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+@Configuration
+@ComponentScan(
+	basePackages = {"org.ngrinder"},
+	useDefaultFilters = false,
+	includeFilters = {@ComponentScan.Filter(type = FilterType.ANNOTATION, value = org.springframework.stereotype.Controller.class)}
+)
+public class WebMvcConfig extends WebMvcConfigurerAdapter {
+
+	@Autowired
+	private LocaleChangeInterceptor localeChangeInterceptor;
+
+	@Autowired
+	private ResourceProperties resourceProperties = new ResourceProperties();
+
+	@Value("${ngrinder.version}")
+	private String ngrinderVersion;
+
+	@Override
+	public void addArgumentResolvers(List<HandlerMethodArgumentResolver> argumentResolvers) {
+		argumentResolvers.add(userHandlerMethodArgumentResolver());
+		argumentResolvers.add(remainedPathMethodArgumentResolver());
+		argumentResolvers.add(pageableHandlerMethodArgumentResolver());
+		super.addArgumentResolvers(argumentResolvers);
+	}
+
+	@Bean
+	public PageableHandlerMethodArgumentResolver pageableHandlerMethodArgumentResolver() {
+		PageableHandlerMethodArgumentResolver pageableHandlerMethodArgumentResolver = new PageableHandlerMethodArgumentResolver();
+		pageableHandlerMethodArgumentResolver.setPrefix("page.");
+		return pageableHandlerMethodArgumentResolver;
+	}
+
+	@Bean
+	public RemainedPathMethodArgumentResolver remainedPathMethodArgumentResolver() {
+		return new RemainedPathMethodArgumentResolver();
+	}
+
+	@Bean
+	public UserHandlerMethodArgumentResolver userHandlerMethodArgumentResolver() {
+		return new UserHandlerMethodArgumentResolver();
+	}
+
+	@Override
+	public void configureHandlerExceptionResolvers(List<HandlerExceptionResolver> exceptionResolvers) {
+		exceptionResolvers.add(apiExceptionHandlerResolver());
+		exceptionResolvers.add(exceptionHandlerExceptionResolver());
+	}
+
+	@Bean
+	public ApiExceptionHandlerResolver apiExceptionHandlerResolver() {
+		ApiExceptionHandlerResolver apiExceptionHandlerResolver = new ApiExceptionHandlerResolver();
+		apiExceptionHandlerResolver.setOrder(-1);
+		return apiExceptionHandlerResolver;
+	}
+
+	@Bean
+	public ExceptionHandlerExceptionResolver exceptionHandlerExceptionResolver() {
+		ExceptionHandlerExceptionResolver exceptionHandlerExceptionResolver = new ExceptionHandlerExceptionResolver();
+		exceptionHandlerExceptionResolver.setOrder(1);
+		return exceptionHandlerExceptionResolver;
+	}
+
+	@Override
+	public void addInterceptors(InterceptorRegistry registry) {
+		registry.addInterceptor(localeChangeInterceptor);
+	}
+
+	@Bean
+	public MultipartResolver multipartResolver() {
+		CommonsMultipartResolver commonsMultipartResolver = new CommonsMultipartResolver();
+		commonsMultipartResolver.setMaxUploadSize(50000000);
+		commonsMultipartResolver.setDefaultEncoding("utf-8");
+		return commonsMultipartResolver;
```

Comment:
I think this can be set by spring application.properties without creating bean explicitly.